RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerMultiFit.cc

```diff
@@ -514,6 +513,256 @@ EcalUncalibRecHitWorkerMultiFit::run( const edm::Event & evt,
+                      float sigmat        = std::sqrt( nterm*nterm  + cterm*cterm   );
+                      if ( ( correctedTime > sigmat*outOfTimeThreshP )   ||
+                           ( correctedTime < -sigmat*outOfTimeThreshM )) 
+                        {
+                            uncalibRecHit.setFlagBit( EcalUncalibratedRecHit::kOutOfTime );
+                        }
+                    }
+
+                }
+            } else if (timealgo_ == weightsMethod) {
+                //  weights method on the PU subtracted pulse shape
+                std::vector<double> amplitudes;
+                for(unsigned int ibx=0; ibx<activeBX.size(); ++ibx) amplitudes.push_back(uncalibRecHit.outOfTimeAmplitude(ibx));
+                
+                EcalTBWeights::EcalTDCId tdcid(1);
+                EcalTBWeights::EcalTBWeightMap const & wgtsMap = wgts->getMap();
+                EcalTBWeights::EcalTBWeightMap::const_iterator wit;
+                wit = wgtsMap.find( std::make_pair(*gid,tdcid) );
+                if( wit == wgtsMap.end() ) {
+                    edm::LogError("EcalUncalibRecHitError") << "No weights found for EcalGroupId: " 
+                                                            << gid->id() << " and  EcalTDCId: " << tdcid
+                                                            << "\n  skipping digi with id: " << detid.rawId();
+                    result.pop_back();
+                    continue;
+                }
+                const EcalWeightSet& wset = wit->second; // this is the EcalWeightSet
+                
+                const EcalWeightSet::EcalWeightMatrix& mat1 = wset.getWeightsBeforeGainSwitch();
+                const EcalWeightSet::EcalWeightMatrix& mat2 = wset.getWeightsAfterGainSwitch();
+                
+                weights[0] = &mat1;
+                weights[1] = &mat2;
+                
+                double timerh;
+                if (detid.subdetId()==EcalEndcap) { 
+                    timerh = weightsMethod_endcap_.time( *itdg, amplitudes, aped, aGain, fullpulse, weights);
+                } else {
+                    timerh = weightsMethod_barrel_.time( *itdg, amplitudes, aped, aGain, fullpulse, weights);
+                }
+                uncalibRecHit.setJitter( timerh );
+                uncalibRecHit.setJitterError( 0. ); // not computed with weights
+            }  else { // no time method;
+                uncalibRecHit.setJitter( 0. );
+                uncalibRecHit.setJitterError( 0. );                  
+            }
+        }
+
+	// set flags if gain switch has occurred
+        auto & uncalibRecHit = result.back();
+	if( ((EcalDataFrame)(*itdg)).hasSwitchToGain6()  ) uncalibRecHit.setFlagBit( EcalUncalibratedRecHit::kHasSwitchToGain6 );
+	if( ((EcalDataFrame)(*itdg)).hasSwitchToGain1()  ) uncalibRecHit.setFlagBit( EcalUncalibratedRecHit::kHasSwitchToGain1 );
+
+        // put the recHit in the collection
+        //result.push_back( uncalibRecHit );
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant.
In this case it even seems confusing left as is